openslides_backend/action/actions/speaker/delete.py

```diff
@@ -34,3 +40,30 @@ def check_permissions(self, instance: dict[str, Any]) -> None:
             if meeting_user.get("user_id") == self.user_id and not len(restricted):
                 return
         super().check_permissions(instance)
+
+    def get_updated_instances(self, action_data: ActionData) -> ActionData:
+        self.end_time = round(time())
```

Comment:
Seems that this set's always the end_time and will overwrite an already closed speaker

Reply:
1. This is a delete function. Any speaker whose id is given will be deleted in the same step.
2. The end_time variable is looked up only in line 67, for the expressed purpose of being able to calculate the correct `remaining_time` when stopping the structure level countdown.
3. This mirrors the functionality in `speaker.end_speech`.

Reply:
IMO you don't need to calculate the remaining time, if the end_time was set before.